exec/java-exec/src/main/java/org/apache/drill/exec/resolver/ResolverTypePrecedence.java

```diff
@@ -78,6 +78,9 @@
     precedenceMap.put(MinorType.INTERVALYEAR, i+= 2);
     precedenceMap.put(MinorType.INTERVAL, i+= 2);
     precedenceMap.put(MinorType.UNION, i += 2);
+    precedenceMap.put(MinorType.MAP, i += 2);
```

Comment:
It seems you meant "than Union".
Devs agreed with the place in the end of precedenceMap just before Union
Updated.